File: WebClient SDK/src/default/onlyOffice/registrator.ts
import { ExtensionContainer, ExtensionStage } from 'tessa/extensions';
import { OnlyOfficeFileExtension } from './onlyOfficeFileExtension';
import { OnlyOfficeInitializationApplicationExtension } from './onlyOfficeInitializationApplicationExtension';
import { OnlyOfficeFileControlExtension } from './onlyOfficeFileControlExtension';
import { OnlyOfficeCardUIExtension } from './onlyOfficeCardUIExtension';

ExtensionContainer.instance.registerExtension({
  extension: OnlyOfficeFileExtension,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});

ExtensionContainer.instance.registerExtension({
  extension: OnlyOfficeInitializationApplicationExtension,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});

ExtensionContainer.instance.registerExtension({
  extension: OnlyOfficeFileControlExtension,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});

ExtensionContainer.instance.registerExtension({
  extension: OnlyOfficeCardUIExtension,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});

File: Extensions/Tessa.Extensions.Default.Server/Notices/Registrator.cs
using Tessa.Cards.Caching;
using Tessa.Cards.Extensions;
using Tessa.Extensions.Default.Shared.Notices;
using Tessa.Notices;
using Tessa.Platform.Data;
using Tessa.Platform.Licensing;
using Tessa.Platform.Runtime;
using Unity;
using Unity.Injection;
using Unity.Lifetime;

namespace Tessa.Extensions.Default.Server.Notices
{
    [Registrator]
    public sealed class Registrator : RegistratorBase
    {
        public override void RegisterUnity()
        {
            this.UnityContainer
                .RegisterType<INotificationResolver, NotificationResolver>(
                    new ContainerControlledLifetimeManager(),
                    new InjectionConstructor(
                        new ResolvedParameter<IMailService>(MailServiceNames.Default),
                        new ResolvedParameter<IMailService>(MailServiceNames.WithoutTransaction),
                        typeof(IDbScope),
                        typeof(ISession),
                        typeof(ICardCache),
                        typeof(IUnityContainer),
                        typeof(ILicenseManager)))

                .RegisterType<NotificationStoreExtension>(new ContainerControlledLifetimeManager())
                .RegisterType<NotificationGetExtension>(new ContainerControlledLifetimeManager())

                // .RegisterType<INotificationSender, CustomNotificationSender>(
                //     CustomNotification.Key,
                //     new ContainerControlledLifetimeManager())

                .RegisterType<INotificationDefaultLanguagePicker, KrNotificationDefaultLanguagePicker>(new ContainerControlledLifetimeManager())
                .RegisterType<INotificationSubscriptionPermissionManager, KrNotificationSubscriptionPermissionManagerServer>(new ContainerControlledLifetimeManager())
                ;
        }


        public override void RegisterExtensions(IExtensionContainer extensionContainer)
        {
            extensionContainer
                .RegisterExtension<ICardStoreExtension, NotificationStoreExtension>(x => x
                    .WithOrder(ExtensionStage.AfterPlatform, 1)
                    .WithUnity(this.UnityContainer))
                .RegisterExtension<ICardGetExtension, NotificationGetExtension>(x => x
                    .WithOrder(ExtensionStage.AfterPlatform, 1)
                    .WithUnity(this.UnityContainer))
                ;
        }
    }
}
